### ISSUE TITLE
3.1.1 - reactor ConsumerBeanAutoConfiguration initialized twice

### ISSUE THREAD

**Issue by zyro23** [grails/grails-core#9695]
upgrading to 3.1.1, my `@Selector` methods started getting invoked twice for each event.

debugging into the bean initialization, it seems the `reactor-spring` `ConsumerBeanAutoConfiguration` is configured twice. Once from `EventBusGrailsPlugin`, once from `ReactorAutoConfiguration` (which applies `@EnableReactor`).

* https://github.com/grails/grails-core/blob/master/grails-plugin-events/src/main/groovy/org/grails/plugins/events/EventBusGrailsPlugin.groovy#L48
* https://github.com/spring-projects/spring-boot/blob/master/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/reactor/ReactorAutoConfiguration.java#L49

root cause may be that boot-1.3.x is now using reactor-2.x, too - however i am not sure why the `@ConditionalOnMissingBean` checks in `ReactorAutoConfiguration` do not prevent auto-configuration. For `eventBus` it could be explained b/c grails is using a `FactoryBean`, but at least the `Environment` one applying `@EnableReactor` should not get configured? 

workaround for now:
```groovy
@EnableAutoConfiguration(exclude = [ReactorAutoConfiguration])
class Application extends GrailsAutoConfiguration { ... }
```

**Comment by hakanernam** [comment]
This issue has been present since at least 3.1.0.M3. Please see (also for another workaround):

https://github.com/grails/grails-core/issues/9490

**Comment by ctoestreich** [comment]
:+1:

**Comment by saboya** [comment]
Has this really been fixed? I tested with Grails 3.0.17 and 3.1.6 and I still have this issue. And I also tested both workarounds with no success.
